ember-toucan-core/src/components/form/checkbox-group-field.ts

```diff
@@ -0,0 +1,82 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+import CheckboxFieldComponent from './checkbox-field';
+
+import type { WithBoundArgs } from '@glint/template';
+
+export interface ToucanFormCheckboxGroupFieldComponentSignature {
+  Element: HTMLFieldSetElement;
+  Args: {
+    /**
+     * Provide a string to this argument to render an error message and apply error styling to the Field.
+     */
+    error?: string;
+
+    /**
+     * Provide a string to this argument to render a hint message to help describe the control.
+     */
+    hint?: string;
+
+    /**
+     * Sets the disabled attribute on the fieldset.
+     */
+    isDisabled?: boolean;
+
+    /**
+     * Provide a string to this argument to render inside of the label tag.
+     */
+    label: string;
+
+    /**
+     * Sets the name attribute of the checkboxes. A string specifying a name for the input control. This name is submitted along with the control's value when the form data is submitted.
+     *
+     * @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name
+     */
+    name: string;
+
+    /**
+     * The function called when a checkbox element is clicked.
+     */
+    onChange?: (selectedValues: Array<string>, e: Event | InputEvent) => void;
+
+    /**
+     * A test selector for targeting the root element of the field. In this case, the wrapping div element.
+     */
+    rootTestSelector?: string;
+
+    /**
+     * The currently selected checkbox elements. The elements with the matching `@option` / underlying value attribute will be checked.
+     */
+    value?: Array<string>;
+  };
+  Blocks: {
+    default: [
+      {
+        CheckboxField: WithBoundArgs<
+          typeof CheckboxFieldComponent,
+          'isDisabled' | 'name' | 'value'
+        >;
+      }
+    ];
+  };
+}
+
+export default class ToucanFormCheckboxGroupFieldComponent extends Component<ToucanFormCheckboxGroupFieldComponentSignature> {
+  CheckboxFieldComponent = CheckboxFieldComponent;
+
+  @action
+  handleInput(_: boolean, e: Event | InputEvent): void {
+    let value = this.args.value ? [...this.args.value] : [];
+
+    let target = e.target as HTMLInputElement;
+
+    if (value?.includes(target.value)) {
+      value = value.filter((val) => val !== target.value);
+    } else {
+      value.push(target.value);
+    }
+
+    this.args.onChange?.(value, e);
```

Comment:
Simon and I talked about maybe making the first argument to `onChange` a `Map`/POJO that could look like:

```
{ 'option-1': true, 'option-2': false, 'option-3: false }
```

After looking at existing uses, it looks like we expect this to be an array of the selected items (as done in this PR!).  Due to that, we punted on this; however, as we start integrating it more we may switch to something like this if it makes more sense.

Reply:
Would we need to modify `ember-headless-form` to yield the value/values?

https://github.com/CrowdStrike/ember-headless-form/blob/e6398198dfa1a1d9d51cf439c7eb67fc85d1555d/docs/usage/controls.md#radio-group

It looks like it yields the rawErrors now but not the rawValue if I am reading this correctly: https://github.com/CrowdStrike/ember-headless-form/blob/main/packages/ember-headless-form/src/components/headless-form.hbs#L31